Assets/Scripts/Core/Tools/TrapTrigger.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using Player;
+using UnityEngine;
+
+namespace Core.Tools
+{
+    public class TrapTrigger : MonoBehaviour
+    {
+        [SerializeField] private PlayerEntity _playerEntity;
+        [SerializeField] private float _damage;
+        
+        private void OnTriggerStay2D(Collider2D other)
+        {
+            _playerEntity.Hit(_damage);
```

Comment:
1. What if other collider will enter trigger collider? Imho better aproach would be getting PlayerEntity from collider(in OnTriggerEnter2D)

2. OnTriggerStay will be triggered not every fixed update if rigid body is not set to Never Sleep. Esentially it means that damage is dependant on the frame rate